src/main/java/br/com/digituz/mailer/service/EmailService.java

```diff
@@ -19,13 +28,59 @@
 	@Autowired
 	private EmailRepository emailRepository;
 
+	@Autowired
+	private JavaMailSender javaMailSender;
+
+	@Value("${spring.mail.username}")
+	private String sender;
+
+	@Value("${mailer.replyTo}")
+	private String replyTo;
+
 	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
 	public void saveEmails(Email email) {
 		this.emailRepository.save(email);
 		logger.info("Email saved");
 	}
 
+	public void sendEmails() throws MessagingException {
+
+		MimeMessage mimeMessage = javaMailSender.createMimeMessage();
+
+		MimeMessageHelper helper = new MimeMessageHelper(mimeMessage, true);
+
+		emailsAll()
```

Comment:
A ideia é criar um query com o spring-data passando na cláusula where as condições necessárias para retornar apenas os emails que devem ser enviados. Ou seja, tu tens que filtrar (clasula where) por status = `NEW` ou status = `ERROR`. Mas, lembrando, `ERROR` somente quando a contagem (attempts) for menor que 5.